java/code/src/com/redhat/rhn/taskomatic/task/HubReportDbUpdateDriver.java

```diff
@@ -41,51 +37,67 @@
 public class HubReportDbUpdateDriver implements QueueDriver<MgrServerInfo> {
 
     private Logger log;
-    private static AtomicBoolean hasCandidates = new AtomicBoolean(true);
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
     public void setLogger(Logger loggerIn) {
         this.log = loggerIn;
     }
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
     public Logger getLogger() {
         return log;
     }
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
     public List<MgrServerInfo> getCandidates() {
         CriteriaBuilder builder = HibernateFactory.getSession().getCriteriaBuilder();
         CriteriaQuery<MgrServerInfo> criteria = builder.createQuery(MgrServerInfo.class);
         criteria.from(MgrServerInfo.class);
         Query<MgrServerInfo> query = HibernateFactory.getSession().createQuery(criteria);
         List<MgrServerInfo> mgrServerInfos = query.list();
-        return mgrServerInfos.stream().filter(info ->
-                Optional.ofNullable(info.getReportDbLastSynced())
-                        .map(time -> time.toInstant().plus(24, ChronoUnit.HOURS).isBefore(Instant.now()))
-                        .orElse(true)
-        ).map(e -> {
+        return mgrServerInfos.stream().map(e -> {
```

Comment:
What happens when this task is still running? Can we detect this and prevent duplicate execution?

Reply:
Still testing and trying to understand how it works, but avoid duplicate execution is the main characteristic of `RhnQueueJob` . From comment:
```
Custom Quartz Job implementation which only allows one thread to run at a time. All other threads return without performing any work. This policy was chosen instead of blocking so as to reduce threading problems inside Quartz itself.
```
I'll verify the correct behaviour with some tests.